### PR TITLE
ceph.spec.in: fix path for mib file and properly mark in %files

### PR DESCRIPTION
**Pull request by bigjust**
Fixes typos introduced in https://github.com/ceph/ceph/pull/46918

Signed-off-by: Justin Caratzas <jcaratza@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [x] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by djgalloway** [comment]
If you push this to ceph-ci.git, it will trigger a build and we can see if rpmbuild will run this time.

**Review by tchaikov — APPROVED** [review]
as long as it build on jenkins -- a link to shaman build job would suffice.

**Comment by tchaikov** [comment]
being built at https://shaman.ceph.com/builds/ceph/ci-fix-mib/fddb7dd2b038d52489ec74aefa03cce5dc435e77/

**Comment by djgalloway** [comment]
Getting further.

```
Processing files: ceph-mib-17.0.0-13567.gfddb7dd2.el8.noarch
warning: Ignoring invalid regex ^%{_scl_prefix}/.*|%{_root_sysconfdir}/rpm/macros.gcc-toolset-11-config$
warning: Ignoring invalid regex ^%{_root_sysconfdir}/rpm/macros.gcc-toolset-11-config$
Provides: ceph-mib = 2:17.0.0-13567.gfddb7dd2.el8 config(ceph-mib) = 2:17.0.0-13567.gfddb7dd2.el8
Requires(rpmlib): rpmlib(CompressedFileNames) <= 3.0.4-1 rpmlib(FileDigests) <= 4.6.0-1 rpmlib(PayloadFilesHavePrefix) <= 4.0-1
```

**Comment by djgalloway** [comment]
Built. https://jenkins.ceph.com/job/ceph-dev-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos8,DIST=centos8,MACHINE_SIZE=gigantic/50004/

@tchaikov I don't suppose a teuthology run would be worthwhile here since ceph-mib isn't a dependency of any other package?

**Comment by djgalloway** [comment]
> I'll schedule an rbd job when the build completes and merge.

Looks like you need to reschedule with more jobs.  The two that got picked were Ubuntu.

**Comment by idryomov** [comment]
> Looks like you need to reschedule with more jobs. The two that got picked were Ubuntu.

Those two are for a different PR.  The repos for this branch aren't ready yet.

**Comment by djgalloway** [comment]
> Those two are for a different PR.

Ah, sorry.

> The repos for this branch aren't ready yet.

https://shaman.ceph.com/repos/ceph/ci-fix-mib/fddb7dd2b038d52489ec74aefa03cce5dc435e77/ just finished up

File: ceph.spec.in [hunk 1]
```diff
@@ -2584,6 +2584,6 @@ exit 0
 
 %files mib
 %attr(0755,root,root) %dir %{_datadir}/snmp
-%dir %{_datadir}/snmp/mibs
+%config %{_datadir}/snmp/mibs
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply ceph.spec.in:2587]: this directory belongs to `net-snmp`. see https://src.fedoraproject.org/rpms/net-snmp/blob/rawhide/f/net-snmp.spec#_493, so yeah, `%dir %{_datadir}/snmp/mibs` was wrong.

but `%config %{_datadir}/snmp/mibs` is not correct, i am afraid.

**Comment by tchaikov** [comment]
> Built. https://jenkins.ceph.com/job/ceph-dev-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos8,DIST=centos8,MACHINE_SIZE=gigantic/50004/
> 
> @tchaikov I don't suppose a teuthology run would be worthwhile here since ceph-mib isn't a dependency of any other package?

yeah. but better safe than sorry. =)

File: ceph.spec.in (hunk 1 above)
[the remark this answers is not in the document]
Comment (bigjust) [reply ceph.spec.in:2587]: Originally i had the location as /usr/share/snmp/mib, which wouldn't conflict with snmp/mibs/ but might be confusing? Maybe /usr/share/snmp/ceph-mib would be better? @pcuzner do you have any thoughts?

[the remark this answers is not in the document]
Comment (idryomov) [reply ceph.spec.in:2587]: > but we can fix it later.

Yeah, unbreaking centos builds is the priority.

**Comment by idryomov** [comment]
All green:

https://pulpito.ceph.com/dis-2022-07-15_15:55:23-rbd-ci-fix-mib-distro-default-smithi/

**Review by idryomov — APPROVED** [review]

**Pull request merged**